schemas/2.2.0.json

```diff
@@ -496,6 +496,14 @@
           "type": "string",
           "description": "A description of the channel."
         },
+        "servers": {
+          "type": "array",
+          "description": "The names of the servers on which this channel is available. If absent or empty then this channel must be available on all servers.",
+          "items": {
+            "type": "string"
+          },
+          "uniqueItems": true
+        },
```

Comment:
I think that `minItems: 1` should be also included. For what someone want to define empty array?

Reply:
@magicmatatjahu : we could do that, but the spec defines `servers` as follows: 

> The servers on which this channel is available, specified as an optional unordered list of names (string keys) of Server Objects defined in the Servers Object (a map). If `servers` is absent or empty then this channel must be available on all servers defined in the Servers Object.

This is also the `description` used here in the JSON-Schema.

So we could of course change this, but my idea was:
1. an absent `servers` field should behave like an empty `servers` list (it's also like that in the JS parser: if `servers` is not present the `servers()` function returns an empty array.

Reply:
I know situation in the parser, if no one has provided servers, it is treated as an empty array :) And also I read the description for this field. For me it is not a must have but rather a loose remark. If others see the current scheme as correct, I have no problem with that. I'm only surprised by the case where someone has to provide an empty array, I only see a case with automatic schema generation from code-first perspective.

Reply:
to think it through:
1. an **absent** `servers` field means that the channel is defined in **all servers**
1.1. in the JS parser: `hasServers()` returns `false` and `servers()` returns an empty string array
2. an **empty** `servers` field is **prohibited** in JSON-Schema
2.1. in the JS parser (this is only possible if the AsyncAPI doc is invalid against the JSON-Schema): just like when there is no field: `hasServers()` returns `false` and `servers()` returns an empty string array
3. a `servers` field with **one or more string names** means that the channel is defined only in the **servers with those names**
3.1. in the JS parser: `hasServers()` returns `true` and `servers()` returns a string array with the names

This is also consistent, i think: shall i implement this?

Reply:
But some fields in the parser return for method `has*` as true even though it is an empty array. From what I remember in other places in JSON Schema we also don't have `minItems: 1` defined where it should, so I'd rather be consistent than do otherwise for this case.

Reply:
so we'll leave it as it is and i'll mark this as resolved - thank you all!

Reply:
marking this as resolved as per Fran's comment.